common/djangoapps/course_modes/models.py

```diff
@@ -317,8 +317,8 @@ def can_auto_enroll(cls, course_id, modes_dict=None):
         if modes_dict is None:
             modes_dict = cls.modes_for_course_dict(course_id)
 
-        # Professional mode courses are always behind a paywall
-        if "professional" in modes_dict:
+        # Professional and no-id-professional mode courses are always behind a paywall
+        if "professional" in modes_dict or CourseMode.NO_ID_PROFESSIONAL_MODE[0] in modes_dict:
```

Comment:
Could you please rename this variable `NO_ID_PROFESSIONAL_MODES` (with an 's' at the end)?  Otherwise, it's unclear that it's a list.